Add post-graduation checklist

tobie opened the pull request:
* Include formal IP transfers and related tooling, which cannot happen
  before project has formally graduated from incubation
* Modify onboarding checklist so that IP related tasks are limited to
  listing related assets and committing to post-graduation transfer.
* Use the term trademark instead of logomark.
* Update related issue template
* Create additional issue template for post-graduation checklist.

Closes #378, closes #380.

tobie commented:
Happy to submit this more formally as a proposal if necessary. I just wanted to document this as I was moving through the incubation process with the AMP project and figured a PR was the easiest and most actionable way to do so.

ljharb commented on .github/ISSUE_TEMPLATE/00-project-onboarding-checklist-template.md:
```diff
@@ -9,22 +9,22 @@ assignees: ''
 - [ ] Add OpenJS Foundation logo to project website
 - [ ] Add Project logo to OpenJS Foundation website; update PROJECTS.md file
-- [ ] Transfer logomark to the OpenJS Foundation
+- [ ] List trademarks the project commits to transfer to the OpenJS Foundation following graduation
```
```suggestion
- [ ] List relevant trademarks, indicating which one(s) the project commits to transfer to the OpenJS Foundation following graduation
```

ljharb commented on .github/ISSUE_TEMPLATE/00-project-onboarding-checklist-template.md:
```diff
@@ -9,22 +9,22 @@ assignees: ''
 - [ ] Adopt the OpenJS Foundation Code of Conduct
 - [ ] Update project CoC reporting methods to include OpenJS Foundation escalation path
-- [ ] Transfer official domains to OpenJS Foundation
+- [ ] List official domains that the project commits to transfer to the OpenJS Foundation following graduation
```
```suggestion
- [ ] List official domains, indicating which one(s) the project commits to transfer to the OpenJS Foundation following graduation
```

ljharb commented on .github/ISSUE_TEMPLATE/00-project-onboarding-checklist-template.md:
```diff
@@ -9,22 +9,22 @@ assignees: ''
+- [ ] List official domains that the project commits to transfer to the OpenJS Foundation following graduation
 - [ ] Identify and document other core project infrastructure
-- [ ] If choosing to use a Contributor License Agreement (CLA) or Developer Certificate of Origin (DCO), make selection and implement appropriate tool
+- [ ] If choosing to use a Contributor License Agreement (CLA) or Developer Certificate of Origin (DCO), make selection
```
```suggestion
- [ ] If choosing to use a Contributor License Agreement (CLA) or Developer Certificate of Origin (DCO), choose one, or "None"
```

brianwarner replied:
Per the IP policy, projects should implement the DCO.  The CLA is optional on a per-project basis, though.

ljharb replied:
Why is this the IP policy? https://ben.balter.com/2018/01/02/why-you-probably-shouldnt-add-a-cla-to-your-open-source-project/ seems relevant.

brianwarner replied:
My understanding is that the JSF required all projects to use a CLA (@kborchers may be able to shed some light on the roots of that policy).  By contrast, the Node.js Foundation did not.

The IP Policy strikes a balance between the two by making the CLA optional going forward.  Projects who have a reason to use it may continue to do so, whereas those who either no longer wish to use it or who never had it in the first place can just use DCO.

ljharb replied:
From what https://developercertificate.org implies, this is already how github works implicitly. What does the DCO requirement impose on contributors, maintainers, and projects?

tobie replied:
Suggested a session on these IP-related issues at the collab summit next week [here](https://docs.google.com/document/d/1hdyWxXjCXUe1iEUqaQE8jvt38N6ckQomxTCtgkdisnU/edit). Additional subtopics welcomed.

tobie replied:
So the current IP policy won't let us choose "none" as an option. I won't be merging that suggestion, @ljharb, and suggest we continue this discussion in https://github.com/openjs-foundation/cross-project-council/issues/387.

mhdawson commented:
Based on past process we'll need board approval to land (there is also discussion about relaxing this but it has not concluded yet).

mhdawson commented:
@tobie thanks for the update.

mhdawson commented:
@tobie looks like next step is for you to address some comments and then I'll take it to the board for approval.

mhdawson commented:
Thinking further I think this also needs more approvals from the @openjs-foundation/cpc  based on :

Pull requests that change the charter of the CPC must meet the following conditions in addition to the ones listed for changing CPC governance:

* The text of the PR must be approved by a simple majority of the voting members.
* The text of the PR must be approved by the board.

brianwarner commented on .github/ISSUE_TEMPLATE/00-project-onboarding-checklist-template.md:
```diff
@@ -9,22 +9,22 @@ assignees: ''
+- [ ] If choosing to use a Contributor License Agreement (CLA) or Developer Certificate of Origin (DCO), make selection
 - [ ] Add or Update Governance.md document (required for Impact stage)
 - [ ] Confirm required files in place (CODE_OF_CONDUCT.md, LICENSE.md)
 - [ ] Project Charter is published on website or github
```
- [ ] Add [SPDX short IDs to files](https://spdx.org/ids-how) (optional, but helpful)

tobie replied:
To _all_ files?

ljharb replied:
hopefully not; a SPDX declaration in `package.json` should implicitly cover every file in that package, just like a LICENSE file (that github can detect) should implicitly cover every file the repo.

brianwarner replied:
It's not a requirement, but adding it to each file is a [best practice based upon the Kernel](https://www.kernel.org/doc/html/v4.18/process/license-rules.html).  Think of the guidance this way: You don't need to do it (note the "optional") but say you were modifying a file anyhow and saw it wasn't there, it would be a good idea to add it.

The idea of short-form identifiers is that they maintain the license notice even if the file is cherry-picked out of the project.

tobie replied:
I feel like we should more opinionated about this and go for minimal legal requirements everywhere. This just adds friction and noise otherwise.

ljharb replied:
That's a best practice for the linux kernel, which isn't what i'd call an exemplar for any part of the JS ecosystem (linux is also changed with .patch files, which isn't how much else operates). I very much think we should strive for not just minimal requirements, but minimal recommendations.

I think it is a *bad* idea to add it, because it's extra redundant noise; in practice, files don't exist in isolation from npm packages or from git repos.

tobie replied:
> I think it is a bad idea to add it, because it's extra redundant noise; in practice, files don't exist in isolation from npm packages or from git repos.

Right. This is I think the point I was trying to make. Can we make OpenJSF as dev-friendly as possible and only recommend the minimal necessary legal requirements. The goal should be to remove everything that isn't 100% necessary.

brianwarner replied:
Thank you for both of your comments.

I'd like to suggest we take a step back and recenter on the original comment.  The intent of this change was to document something which already exists in the OpenJS Foundation IP policy.  However, it doesn't necessarily need to be in this particular list, so if it will add more confusion than clarity, I'd have no objections to removing the item from the checklist.

tobie replied:
Yes. Agreed.

I feel like the reason we're having this conversation here is that there was no room (afaik) for discussing the concrete implementation and day-to-day operations impact of the IP policy. Maybe that's something worth discussing next week at the summit?

tobie replied:
Suggested a session on these IP-related issues at the collab summit next week [here](https://docs.google.com/document/d/1hdyWxXjCXUe1iEUqaQE8jvt38N6ckQomxTCtgkdisnU/edit). Additional subtopics welcomed.

tobie replied:
Seems we don't have consensus around this guidance suggestion and have agreement not to include it here for the time being. Marking as resolved.

MylesBorins commented:
The other issue template for onboarding got moved to another repo. Should this move there too?

tobie commented:
> The other issue template for onboarding got moved to another repo. Should this move there too?

Do you want to approve it here first?

mhdawson commented:
@tobie I'm good either way, the template don't need Board approval so if we keep them in this PR and get approval then we can move them easily afterwards.

tobie commented:
@mhdawson easier for me to merge this first and then move. SSD died. I'm currently using a Chromebook so pretty much limited to GH's web interface.

mhdawson commented:
@openjs-foundation/cpc, @tobie  if we can agree and get this ready to land we could possibly get a vote on it by the board this week.

mhdawson commented:
@tobie as per https://github.com/openjs-foundation/cross-project-council/blob/master/GOVERNANCE.md#merging-prs-into-this-repository it would need at least 1/2 of the voting members to approve.

mhdawson commented:
@tobie and of course the conflicts in the PR would need to be fixed up as well.

maddhruv commented on PROJECT_PROGRESSION.md:
```diff
@@ -91,23 +91,33 @@ This is an informational checklist to help projects onboard into the OpenJS Foun
+
+<!--If this checklist is updated please ensure `.github/ISSUE_TEMPLATE/project-post-graduation-checklist-template.md` is updated as well -->
+
+This is an checklist to help projects complete the post-graduation tasks they committed to during onboarding - tasks we will complete together after your project has graduated from the incubation process. If you have any questions or need help, the OpenJS Foundation CPC is available to assist.
```
```suggestion
This is a checklist to help projects complete the post-graduation tasks they committed to during onboarding - tasks we will complete together after your project has graduated from the incubation process. If you have any questions or need help, the OpenJS Foundation CPC is available to assist.
```

nitpick :see_no_evil:

joesepi commented:
9 out of 15 voting members have approved this PR. We should resolve conflicts, merge and then move any documentation that has moved to new repos.

tobie commented:
Updated this issue so that it can now be merged.

Added https://github.com/openjs-foundation/project-onboarding/pull/42 sot that the additional issue template and changes to the pre-existing template are added to the project-onboarding repository.

mhdawson commented:
@tobie thanks. I'll send an email to @brianwarner asking for a Board vote on this which we believe is required for this one.

tobie commented on PROJECT_PROGRESSION.md:
```diff
@@ -86,29 +86,39 @@ The Application Champion is a member of the CPC who commits to do the following:
+
+- [ ] Transfer official domains to OpenJS Foundation
+- [ ] If relevant, implement the appropriate tool to enforce CLA or DCO
+- [ ] Transfer trademarks to the OpenJS Foundation
```
As suggested by @ljharb in https://github.com/openjs-foundation/project-onboarding/pull/42.

```suggestion
- [ ] Transfer previously indicated official domains to OpenJS Foundation, if any
- [ ] If relevant, implement the appropriate tool to enforce CLA or DCO
- [ ] Transfer previously indicated trademarks to the OpenJS Foundation, if any
```

mhdawson commented:
@brianwarner confirmed board vote is complete, landing
